mod/primitives/pkg/hex/string.go

```diff
@@ -48,7 +49,7 @@ func (s *String) UnmarshalText(text []byte) error {
 	str := string(text)
 	err := isValidHex(str)
 	if err != nil {
-		return fmt.Errorf("invalid hex string: %s, error: %w", str, err)
+		return errors.Newf("invalid hex string: %s, error: %w", str, err)
```

Comment:
can you create hard type errors so we can use errors.Is 

also use our mod/errors